src/plugins/dashboard/public/application/hooks/use_dashboard_container.ts

```diff
@@ -106,7 +116,10 @@ export const useDashboardContainer = (
       }
       setDashboardContainer(newContainer);
     })();
-    return () => setDashboardContainer(null);
+    return () => {
+      canceled = true;
+      container?.destroy();
```

Comment:
```suggestion
      canceled = true;
      container?.destroy();
      setDashboardContainer(null);
```

Setting the dashboard container state back to null is important to avoid `embeddable has been destroyed` errors when saving a dashboard as a copy, or initially saving a new dashboard.